pkg/db/db_test.go

```diff
@@ -0,0 +1,435 @@
+		dbname := ""
+		expectedErr := db.ErrDBNameEmpty
+
+		actualPath, err := db.GetDBLoc(dbname)
+
+		require.Error(t, err)
+		require.Equal(t, expectedErr, err)
+		require.Empty(t, actualPath)
+	})
+
+	t.Run("Test getDBLoc with panic", func(t *testing.T) {
+
+	})
+}
+
+func Test_GetRecord(t *testing.T) {
+	t.Run("Test GetRecord", func(t *testing.T) {
+		dbname := db.DBTestName
+		DBLoc, err := db.GetDBLoc(dbname)
+		require.NoError(t, err)
+		defer os.Remove(DBLoc)
+
+		expectedValue := "testValue"
+		key := "testKey"
+		bucketName := "testBucket"
+
+		err = db.WriteToDB(dbname, map[string]string{key: expectedValue}, bucketName)
+		require.NoError(t, err)
+
+		actualValue, err := db.GetRecord(dbname, key, bucketName)
+		require.NoError(t, err)
+		require.Equal(t, expectedValue, actualValue)
+	})
+
+	t.Run("Test GetRecord with empty dbname", func(t *testing.T) {
+		dbname := ""
+		key := "testKey"
+		bucketName := "testBucket"
+		expectedErr := db.ErrDBNameEmpty
+
+		actualValue, err := db.GetRecord(dbname, key, bucketName)
+
+		require.Error(t, err)
+		require.Equal(t, expectedErr, err)
+		require.Empty(t, actualValue)
+	})
+
+	t.Run("Test GetRecord with key not found", func(t *testing.T) {
+		dbname := db.DBTestName
+		DBLoc, err := db.GetDBLoc(dbname)
+		require.NoError(t, err)
+		defer os.Remove(DBLoc)
+		key := "testKey"
+		bucketName := "testBucket"
+		expectedErr := db.ErrKeyNotFound
+
+		err = db.WriteToDB(dbname, map[string]string{}, bucketName)
+		require.NoError(t, err)
+
+		actualValue, err := db.GetRecord(dbname, key, bucketName)
+
+		require.Error(t, err)
+		require.Equal(t, expectedErr, err)
+		require.Empty(t, actualValue)
+	})
+
+	t.Run("Test GetRecord with bucket not found", func(t *testing.T) {
+		dbname := db.DBTestName
+		key := "testKey"
+		bucketName := "testBucket"
+		expectedErr := db.ErrBucketNotFound
+
+		actualValue, err := db.GetRecord(dbname, key, bucketName)
+
+		require.Error(t, err)
+		require.Equal(t, expectedErr, err)
+		require.Empty(t, actualValue)
+	})
+}
+func Test_WriteToDB(t *testing.T) {
+	t.Run("Test WriteToDB", func(t *testing.T) {
+		dbname := db.DBTestName
+		DBLoc, err := db.GetDBLoc(dbname)
+		require.NoError(t, err)
+		defer os.Remove(DBLoc)
+
+		data := map[string]string{
+			"key1": "value1",
+			"key2": "value2",
+			"key3": "value3",
+		}
+		bucketName := "testBucket"
+
+		err = db.WriteToDB(dbname, data, bucketName)
+		require.NoError(t, err)
+
+		// Verify that the data was written correctly
+		db, err := bolt.Open(DBLoc, 0o600, nil)
+		require.NoError(t, err)
+		defer db.Close()
+
+		err = db.View(func(tx *bolt.Tx) error {
+			bucket := tx.Bucket([]byte(bucketName))
+			require.NotNil(t, bucket)
+
+			for k, v := range data {
+				value := bucket.Get([]byte(k))
+				require.Equal(t, []byte(v), value)
+			}
+
+			return nil
+		})
+		require.NoError(t, err)
+	})
+
+	t.Run("Test WriteToDB with empty bucketname", func(t *testing.T) {
+		dbname := db.DBTestName
```

Comment:
You are defining it in almost every single test. Add a const at the beginning of _test package and stop redefining again and again

Reply:
You are talking about what in this part ? i don't get it

Reply:
adding a `const dbname :=  db.DBTestName` at the top of the package

so you don't have to repeat it everywhere. And when you need to set it to empty string, as I think it's the only other use case, you can then use either another variable or using `dbname := ""`